hazelcast/src/main/java/com/hazelcast/map/impl/EntryViews.java

```diff
@@ -58,11 +58,23 @@ public static <K, V> WanMapEntryView<K, V> createWanEntryView(Data key, Data val
                 .withVersion(record.getVersion())
                 .withHits(record.getHits())
                 .withLastAccessTime(record.getLastAccessTime())
-                .withLastUpdateTime(record.getLastUpdateTime())
+                .withLastUpdateTime(calculateLastUpdateTime(record, expiryMetadata))
                 .withCreationTime(record.getCreationTime())
                 .withLastStoredTime(record.getLastStoredTime())
                 .withTtl(expiryMetadata.getTtl())
                 .withMaxIdle(expiryMetadata.getMaxIdle())
                 .withExpirationTime(expiryMetadata.getExpirationTime());
     }
+
+    private static <V> long calculateLastUpdateTime(Record<V> record, ExpiryMetadata expiryMetadata) {
+        if (record.getLastUpdateTime() != Record.UNSET) {
+            return record.getLastUpdateTime();
+        }
+
+        if (expiryMetadata != ExpiryMetadata.NULL) {
+            return expiryMetadata.getLastUpdateTime();
+        }
+
+        return Record.UNSET;
+    }
```

Comment:
Can't we simplify ut further like this?

```suggestion
    private static <V> long calculateLastUpdateTime(Record<V> record, ExpiryMetadata expiryMetadata) {
       if (expiryMetadata != ExpiryMetadata.NULL) {
            return expiryMetadata.getLastUpdateTime();
        }
        
        return record.getLastUpdateTime();
    }
```

Reply:
These will fail if I make this change: https://github.com/hazelcast/hazelcast/pull/23279#issuecomment-1368761967

Reply:
This is probably because we want `EntryProcessor` to touch `lastUpdateTime` but not `expiryTime`.

Reply:
i see, this sounds like an issue. Let's merge this PR and follow the issue here: https://github.com/hazelcast/hazelcast/issues/23340